src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -63,4 +72,19 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
 
 
     }
+
+    /**
+     * This method performs a search operation on a list of movies based on a given search term.
+     * It updates the observableMovies list with the search results.
+     *
+     * @param searchTerm The keyword to search in the movie titles and descriptions.
+     */
+    private void searchKeyword(String searchTerm) {
+        observableMovies.setAll(allMovies);
+        MovieSearchService movieSearchService = new MovieSearchService(observableMovies);
+        Set<Movie> searchResults = new HashSet<>();
+        searchResults.addAll(movieSearchService.searchForMovie(searchTerm));
+        observableMovies.clear();
```

Comment:
I would make a few changes here:
* create a field in the class `private final FilteredList<Movie> filteredMovies = new FilteredList<>(observableMovies);`
* set the `movieListView` to use the `filteredMovies` in the `initialize()` method instead of the `observableList`
* change here to use the FilteredList's `setPredicate` method instead of manipulating the observableMovies list, like this: `filteredMovies.setPredicate(movie -> searchResults.contains(movie));`

Reply:
Updated based on the recommendation